tiledb/sm/misc/comparators.h

```diff
@@ -149,75 +162,60 @@ class GlobalCmp {
    * @param b The second coordinate.
    * @return `true` if `a` precedes `b` and `false` otherwise.
    */
-  bool operator()(const ResultCoords<T>& a, const ResultCoords<T>& b) const {
+  bool operator()(const ResultCoords& a, const ResultCoords& b) const {
     // Compare tile order first
-    auto tile_cmp =
-        domain_->tile_order_cmp_tile_coords<T>(a.tile_coords_, b.tile_coords_);
+    if (tile_order_ == Layout::ROW_MAJOR) {
+      for (unsigned d = 0; d < dim_num_; ++d) {
+        auto res = domain_->tile_order_cmp(d, a.coord(d), b.coord(d));
 
-    if (tile_cmp == -1)
-      return true;
-    if (tile_cmp == 1)
-      return false;
-    // else tile_cmp == 0 --> continue
+        if (res == -1)
+          return true;
+        if (res == 1)
+          return false;
+        // else same tile on dimension d --> continue
+      }
+    } else {  // COL_MAJOR
+      for (unsigned d = dim_num_ - 1;; --d) {
+        auto res = domain_->tile_order_cmp(d, a.coord(d), b.coord(d));
 
-    // Compare cell order
-    auto cell_cmp = domain_->cell_order_cmp(a.coords_, b.coords_);
-    return cell_cmp == -1;
-  }
+        if (res == -1)
+          return true;
+        if (res == 1)
+          return false;
+        // else same tile on dimension d --> continue
 
-  /**
-   * Comparison operator for a vector of integer positions.
-   * Here `buff_` is **not** `nullptr` and a position corresponds to
-   * coordinates in `buff_`.
-   *
-   * @param a The first coordinate position.
-   * @param b The second coordinate position.
-   * @return `true` if coordinates at `a` precedes coordinates at `b`,
-   *     and `false` otherwise.
-   */
-  bool operator()(uint64_t a, uint64_t b) const {
-    // Get coordinates
-    const T* coords_a = &buff_[a * dim_num_];
-    const T* coords_b = &buff_[b * dim_num_];
-    // Compare tile order first
-    auto tile_cmp = domain_->tile_order_cmp<T>(coords_a, coords_b);
-
-    if (tile_cmp == -1)
-      return true;
-    if (tile_cmp == 1)
-      return false;
-    // else tile_cmp == 0 --> continue
+        if (d == 0)
```

Comment:
same thing, let's put this in the for-loop on line 178: `for (unsigned d = dim_num_ - 1; d >= 0; --d) {`